x-pack/filebeat/input/awss3/metrics.go

```diff
@@ -174,6 +174,7 @@ func newInputMetrics(id string, optionalParent *monitoring.Registry, maxWorkers
 		sqsMessagesInflight:                 monitoring.NewUint(reg, "sqs_messages_inflight_gauge"),
 		sqsMessagesReturnedTotal:            monitoring.NewUint(reg, "sqs_messages_returned_total"),
 		sqsMessagesDeletedTotal:             monitoring.NewUint(reg, "sqs_messages_deleted_total"),
+		sqsMessagesWaiting:                  monitoring.NewInt(reg, "sqs_messages_waiting_gauge"),
```

Comment:
The value is initialized upon first use such that the metric is unreported when it cannot be collected (usually due to lack of SQS permissions). If we initialize it from the start then let also apply these changes:

- Initialize the `sqs_messages_waiting_gauge` value to `-1` so that we can distinguish between and no messages waiting (`0`) and never collected / error collecting (`-1`).
- Remove `setSQSMessagesWaiting()` which does lazy init.

Reply:
@andrewkroh 

> Initialize the sqs_messages_waiting_gauge value to -1 so that we can distinguish between and no messages waiting (0) and never collected / error collecting (-1).

Today, the value of `sqs_messages_waiting_gauge` is set every minute in a [pollSQSWaitingMetric](https://github.com/elastic/beats/blob/main/x-pack/filebeat/input/awss3/input.go#L431) function. Do we want to initialize the value to `0` once we receive messages successfully in `SQSReceiver` [somewhere here](https://github.com/elastic/beats/blob/main/x-pack/filebeat/input/awss3/sqs.go#L81) and before we start processing go routines for each SQS Message ?